webapp/src/components/sidebar/sidebarSettingsMenu.tsx

```diff
@@ -4,26 +4,45 @@ import React, {useContext, useState} from 'react'
 import {FormattedMessage, useIntl} from 'react-intl'
 
 import {Archiver} from '../../archiver'
-import {darkTheme, defaultTheme, lightTheme, setTheme, Theme} from '../../theme'
+import {
+    darkTheme,
+    darkThemeName,
+    defaultTheme,
+    defaultThemeName,
+    lightTheme,
+    lightThemeName,
+    setTheme, systemThemeName,
+    Theme,
+} from '../../theme'
 import Menu from '../../widgets/menu'
 import MenuWrapper from '../../widgets/menuWrapper'
 import {SetLanguageContext} from '../../setLanguageContext'
 import {UserSettings} from '../../userSettings'
 
 import './sidebarSettingsMenu.scss'
+import CheckIcon from '../../widgets/icons/check'
 
 type Props = {
     setWhiteLogo: (whiteLogo: boolean) => void
+    activeTheme: string
+    activeLanguage: string
 }
 
 const SidebarSettingsMenu = React.memo((props: Props) => {
     const intl = useIntl()
     const setLanguage = useContext(SetLanguageContext)
 
-    const updateTheme = (theme: Theme | null) => {
+    // we need this as the sidebar doesn't always need to re-render
+    // on theme change. This can cause props and the actual
+    // active theme can go out of sync
+    const [themeName, setThemeName] = useState(props.activeTheme)
+    const [lang, setLang] = useState(props.activeLanguage)
```

Comment:
maybe we can replace this with `useIntl` and use the `intl.locale` to identify the current locale. That way we don't need to recieve or track the the language.

Reply:
On it.

Reply:
Done.